client/client.go

```diff
@@ -1247,6 +1289,207 @@ func (c *Client) runAllocs(update *allocUpdates) {
 	}
 }
 
+// blockForRemoteAlloc blocks until the previous allocation of an allocation has
+// been terminated and migrates the snapshot data
+func (c *Client) blockForRemoteAlloc(alloc *structs.Allocation) {
+	c.logger.Printf("[DEBUG] client: blocking alloc %q for previous allocation %q", alloc.ID, alloc.PreviousAllocation)
+
+	// Removing the allocation from the set of allocs which are currently
+	// undergoing migration
+	defer func() {
+		c.migratingAllocsLock.Lock()
+		delete(c.migratingAllocs, alloc.ID)
+		c.migratingAllocsLock.Unlock()
+	}()
+
+	var err error
+	var prevAlloc *structs.Allocation
+	var index uint64
+	for {
+		// Stopping if the client is shutting down.
+		if c.shutdown {
+			return
+		}
+
+		// get the remote allocation.
+		prevAlloc, index, err = c.getAllocation(alloc.PreviousAllocation, index)
+		if err != nil {
+			c.logger.Printf("[ERROR] client: error getting previous allocation %q for alloc %q: %v",
+				alloc.PreviousAllocation, alloc.ID, err)
+			return
+		}
+
+		// If the previous allocation has been terminated then we stop.
+		if prevAlloc == nil || prevAlloc.Terminated() {
+			break
+		}
+		continue
+	}
+
+	var prevAllocDir *allocdir.AllocDir
+	if prevAlloc != nil {
+		tg := alloc.Job.LookupTaskGroup(alloc.TaskGroup)
+
+		// Migrate the data from remote node if the disk is sticky and migrate
+		// is turned on
+		if tg != nil && tg.EphemeralDisk.Sticky && tg.EphemeralDisk.Migrate {
```

Comment:
Invert the statement so you don't have to indent everything
```
if tg == nil || !tg.EphemeralDisk.Sticky || !tg.EphemeralDisk.Migrate {
   return nil // Assuming my suggestion that you break this up
}
```

Same on the getNode if statement